src/app/modules/submission/__tests__/submission/submission.service.spec.ts

```diff
@@ -152,6 +152,43 @@ describe('submission.service', () => {
         new ConflictError('Some form fields are missing'),
       )
     })
+    it('should allow responses for encrypt mode hidden fields', async () => {
+      // Arrange
+      // Only check for mobile and email fields, since the other fields are
+      // e2e encrypted from the browser.
+      const mobileField = generateDefaultField(BasicField.Mobile)
+      const emailField = generateDefaultField(BasicField.Email)
+      // Add answers to both mobile and email fields
+      const mobileResponse = generateSingleAnswerResponse(
+        mobileField,
+        '+6587654321',
+      )
+      mobileResponse.isVisible = false
```

Comment:
this made me realise that the tests are no longer behaving as intended. the object which we pass into `getProcessedResponses` should be a `FieldResponse`, not some form of `ProcessedResponse`. might it be better to write a `generateFieldResponse` function which doesn't include `isVisible` or `isUserVerified`? or even better, rename `generateSingleAnswerResponse` to `generateProcessedSingleAnswerResponse`?

Reply:
edited